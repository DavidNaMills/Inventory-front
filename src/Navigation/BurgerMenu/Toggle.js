import React from 'react';

const styleObj = {
    backgroundColor: 'rgb(151, 184, 240)',
    borderRadius: '5px'
}

const Toggle = ({ onClick, isOpen = false }) => {


    const style = isOpen
    ? "hamburger hamburger--collapse is-active"
    : "hamburger hamburger--collapse"
    return (
        <button className={style} type="button" style={styleObj} onClick={onClick}>
            <span className={"hamburger-box"}>
                <span className={"hamburger-inner"}></span>
            </span>
        </button>
    )
}

export default Toggle;