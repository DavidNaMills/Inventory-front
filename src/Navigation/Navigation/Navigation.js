import React from 'react';
import classes from './navigation.module.scss';

import { withOrientationChange } from 'react-device-detect'
import NavItem from './NavItem/NavItem';
import HomeIcon from '@material-ui/icons/Home';
import LocationCityIcon from '@material-ui/icons/LocationCity';
import PeopleIcon from '@material-ui/icons/People';
import LocalCafeIcon from '@material-ui/icons/LocalCafe';
import MenuBookIcon from '@material-ui/icons/MenuBook';
import {useLocale} from '../../Context/LocaleContext';

const Navigation = (props) => {
    const locale = useLocale();
    const { userRoutes, isMobile=false, tempClose=null} = props;
    const width = isMobile ? null : (window.innerWidth-20) / userRoutes.length;

    return (
        <div className={isMobile ? classes.navigationMobile : classes.navigation}>
            {
                userRoutes.map((x, y) => {
                    switch(x.to){
                        case '/dashboard':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={HomeIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/products':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={LocalCafeIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/productTypes':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={MenuBookIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/locations':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={LocationCityIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        case '/staff':
                            return <NavItem key={y} to={x.to} label={locale[x.label]} Icon={PeopleIcon} width={width} isMobile={isMobile} tempClose={tempClose}/>
                        default:
                            return <NavItem key={y} to={x.to} label={locale[x.label]} width={width} isMobile={isMobile} tempClose={tempClose}/>
                    }
                })
            }
        </div>
    )
};

export default withOrientationChange(Navigation);