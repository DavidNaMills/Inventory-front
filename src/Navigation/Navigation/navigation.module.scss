@import '../../styles/variables/colors.module.scss';

@mixin navigation {
    display: flex;
    background-color: $nav-back;
    border-bottom: solid $nav-border 3px;
    margin-bottom: 15px;
    z-index: 999;
}

.navigation {
    @include navigation;
    justify-content: space-between;
    top: 0;
    position: sticky;
}

.navigationMobile {
    @include navigation;
    position: absolute;
    width: 100%;
    height: 100vh;
    padding-top: 20px;
    padding-bottom: 20px;
    flex-direction: column;
    // justify-content: space-between;
}
