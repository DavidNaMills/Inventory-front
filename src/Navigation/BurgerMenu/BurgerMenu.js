import React from 'react';
import Toggle from './Toggle';
import Navigation from '../Navigation/Navigation';
import useShowHook from '../../hooks/useShowHook/useShowHook';

const style={
    top: '0',
    position: 'sticky',
    zIndex: '9999'
}

const BurgerMenu = (props) => {
    const {userRoutes, locations} = props;
    const {isShow, reverseBoolean, changeShow} = useShowHook(false);

    return (
        <div style={style}>
            <Toggle onClick={() => reverseBoolean()} isOpen={isShow} />
            {isShow &&
                <Navigation userRoutes={userRoutes} locations={locations} isMobile={true} tempClose={()=>changeShow(false)}/>
            }
        </div>
    )
}

export default BurgerMenu;