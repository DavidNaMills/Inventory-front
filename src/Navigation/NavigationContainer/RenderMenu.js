import React from 'react';
import { isMobileOnly } from 'react-device-detect';

import Navigation from '../Navigation/Navigation';
import BurgerMenu from '../BurgerMenu/BurgerMenu';


const RenderMenu = ({routes, baseId}) => (
    isMobileOnly
        ? <BurgerMenu userRoutes={routes} locations={baseId} />
        : <Navigation userRoutes={routes} locations={baseId} />
);

export default RenderMenu;