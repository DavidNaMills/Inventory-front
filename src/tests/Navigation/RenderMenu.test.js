import React from 'react';
import '../testHelpers/context';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import * as device from 'react-device-detect';

import Navigation from '../../Navigation/Navigation/Navigation';
import BurgerMenu from '../../Navigation/BurgerMenu/BurgerMenu';

import RenderMenu from '../../Navigation/NavigationContainer/RenderMenu';

const routes = [
    { to: '/dashboard', label: 'nav_home' },
    { to: '/products', label: 'nav_prod_man' },
    { to: '/productTypes', label: 'nav_prod_ty' },
    { to: '/locations', label: 'nav_location' },
    { to: '/staff', label: 'nav_staff' },
    { to: '/test', label: 'nav_staff' }
];

const locations = ['123', '456123'];

describe('RenderMenu test suite', () => {
    it('matches snapshot', () => {
        const comp = renderer.create(
            <BrowserRouter>
                <RenderMenu routes={routes} locations={locations} />
            </BrowserRouter>
        );
        const app =comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('renders a <BurgerMenu/> if isMobileOnly === true', ()=>{
        const origValue = device.isMobileOnly;
        device.isMobileOnly = true;
        const wrapper = mount(
        <BrowserRouter>
            <RenderMenu routes={routes} locations={locations} />
        </BrowserRouter>
        )
        expect(wrapper.find(BurgerMenu).length).toBe(1);
        device.isMobileOnly = origValue;
    });

    it('renders a <Navigation/> if isMobileOnly === false', ()=>{
        const wrapper = mount(
        <BrowserRouter>
            <RenderMenu routes={routes} locations={locations} />
        </BrowserRouter>
        )
        expect(wrapper.find(Navigation).length).toBe(1);
    });
});