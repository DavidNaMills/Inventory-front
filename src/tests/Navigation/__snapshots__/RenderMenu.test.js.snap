// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMenu test suite matches snapshot 1`] = `
<div
  className="navigation"
>
  <a
    aria-current={null}
    className="navItem"
    href="/dashboard"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
    Dashboard
  </a>
  <a
    aria-current={null}
    className="navItem"
    href="/products"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 3H4v10c0 2.21 1.79 4 4 4h6c2.21 0 4-1.79 4-4v-3h2c1.11 0 2-.9 2-2V5c0-1.11-.89-2-2-2zm0 5h-2V5h2v3zM4 19h16v2H4z"
      />
    </svg>
    Products
  </a>
  <a
    aria-current={null}
    className="navItem"
    href="/productTypes"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
      />
      <path
        d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
      />
    </svg>
    Product Types
  </a>
  <a
    aria-current={null}
    className="navItem"
    href="/locations"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
      />
    </svg>
    Locations
  </a>
  <a
    aria-current={null}
    className="navItem"
    href="/staff"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
      />
    </svg>
    Staff
  </a>
  <a
    aria-current={null}
    className="navItem"
    href="/test"
    onClick={[Function]}
    style={
      Object {
        "width": 167.33333333333334,
      }
    }
  >
    Staff
  </a>
</div>
`;
